pkg/asset/installconfig/gcp/validation.go

```diff
@@ -219,15 +219,17 @@ func validateMachineNetworksContainIP(fldPath *field.Path, networks []types.Mach
 }
 
 //ValidateEnabledServices gets all the enabled services for a project and validate if any of the required services are not enabled.
+//also warns the user if optional services are not enabled.
 func ValidateEnabledServices(ctx context.Context, client API, project string) error {
-	services := sets.NewString("compute.googleapis.com",
-		"cloudapis.googleapis.com",
+	requiredServices := sets.NewString("compute.googleapis.com",
 		"cloudresourcemanager.googleapis.com",
 		"dns.googleapis.com",
 		"iam.googleapis.com",
 		"iamcredentials.googleapis.com",
+		"deploymentmanager.googleapis.com",
+		"serviceusage.googleapis.com")
```

Comment:
What about `serviceusage`,  how are we determining this is required? Previously it was listed as `optional`...

Reply:
I listed it as optional first but I think it's being used by the installer/google-sdk to get the list of APIs supported? If that is true then this might be the most important one.

Reply:
> I listed it as optional first but I think it's being used by the installer/google-sdk to get the list of APIs supported? If that is true then this might be the most important one for this logic of checking the APIs.

Ah thanks for explaining. This makes sense as to what the API is needed for but I think we should keep is as optional because the API check is optional and will be handled gracefully:

https://github.com/openshift/installer/blob/60ad1fe02453c00f5ff1a33f4e78e5ccc2dee13b/pkg/asset/installconfig/gcp/validation.go#L235-L244

So let's move it back to optional, if you agree.

Reply:
Sounds good to me. I'll change it.